src/openrct2-ui/drawing/engines/opengl/TextureCache.cpp

```diff
@@ -61,7 +61,7 @@ void TextureCache::InvalidateImage(uint32_t image)
     }
 }
 
-BasicTextureInfo TextureCache::GetOrLoadImageTexture(uint32_t image)
+AtlasTextureInfo TextureCache::GetOrLoadImageTexture(uint32_t image)
```

Comment:
Could you elaborate on this?

Reply:
The function actually creates an `AtlasTextureInfo`, not a `BasicTextureInfo`, so returning it by value actually slices it.

Reply:
Please revert this, this is done on purpose for performance reasons.

Reply:
We should add a comment, then.

Reply:
I can revert it, but I agree with Michael that we should leave a comment with a good explanation there, then.

Reply:
Let's do that: revert the change and add a comment: `Note: for performance reasons, this functions returns a BasicTextureInfo instead of an AtlasTextureInfo.`

Reply:
Its also not strictly performance reasons, the data is the internal state of the cache and the callee shouldn't care about that.

Reply:
Done